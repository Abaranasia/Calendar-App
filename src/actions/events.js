import Swal from "sweetalert2";
import { fetchConToken } from "../helpers/fetch"
import { types } from "../types/types"


export const eventStartAddNew = (event) => {
    return async (dispatch, getState) => {

        const { uid, name } = getState().auth;

        try {
            const resp = await fetchConToken('events', event, 'POST');
            const body = await resp.json();
            console.log(body)
            if (body.ok) {
                event.id = body.evento.id;
                event.user = { // el usuario lo podemos sacar del state de redux con getState
                    _id: uid,
                    name: name
                }

                dispatch(eventAddNew(event))
            }
        } catch (error) {
            console.log(error)
            Swal.fire('Error', error, 'error')
        }
    }

}

const eventAddNew = (event) => ({ // AÃ±adir nuevo evento, pero solo al guardar en la BBDD. Sin export por ser interna
    type: types.eventAddNew,
    payload: event
})


export const eventSetActive = (event) => ({ // Establecer como activo un evento (y mostrarlo en el modal)
    type: types.eventSetActive,
    payload: event
})


export const eventClearActiveNote = () => ({ // Elimina el evento activo
    type: types.eventClearActiveEvent
})


export const eventUpdated = (event) => ({ // permite actualizar un evento existente
    type: types.eventUpdated,
    payload: event
})


export const eventDeleted = () => ({ // permite eliminar un evento existente
    type: types.eventDeleted
})