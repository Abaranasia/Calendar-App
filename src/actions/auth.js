import Swal from "sweetalert2";
import { fetchConToken, fetchSinToken } from "../helpers/fetch";
import { types } from "../types/types";
import { eventLogout } from "./events";


export const startLogin = (email, password) => {
    return async (dispatch) => { // Pongo este return porque es una tarea asÃ­ncrona. Dispatch proporcionado por thunk

        // console.log("AUTH: ", email, password);
        const resp = await fetchSinToken('auth', { email, password }, 'POST'); // Nuestro helper sin token
        const body = await resp.json();
        // console.log(body)

        if (body.ok) {
            localStorage.setItem('token', body.token)
            localStorage.setItem('token-init-date', new Date().getTime()) // Momento de inicio del token

            dispatch(login({ // Loguea en redux
                uid: body.uid,
                name: body.name
            }))
        } else {
            Swal.fire('Error', body.msg, 'error')
        }

    }
}


export const startRegister = (name, email, password) => {
    return async (dispatch) => {
        const resp = await fetchSinToken('auth/new', { name, email, password }, 'POST'); // Nuestro helper sin token
        const body = await resp.json();
        // console.log(body)

        if (body.ok) {
            localStorage.setItem('token', body.token)
            localStorage.setItem('token-init-date', new Date().getTime()) // Momento de inicio del token

            dispatch(login({ // Loguea en redux
                uid: body.uid,
                name: body.name
            }))
        } else {
            Swal.fire('Error', body.msg, 'error')
        }
    }
}


export const startChecking = () => { // Comprobamos que se ha renovado el token
    return async (dispatch) => {
        const resp = await fetchConToken('auth/renew', {}, 'GET'); // Nuestro helper sin token
        const body = await resp.json();
        console.log(body)

        if (body.ok) {
            localStorage.setItem('token', body.token)
            localStorage.setItem('token-init-date', new Date().getTime()) // Momento de inicio del token

            dispatch(login({ // Loguea en redux
                uid: body.uid,
                name: body.name
            }))
        } else {
            dispatch(checkingFinish())
            Swal.fire('Error', body.msg, 'error');
        }
    }
}


export const startLogout = () => {
    return (dispatch) => {
        localStorage.clear();
        dispatch(eventLogout()); // limpia evento activo
        dispatch(eventLogout()); // limpia evento activo
        dispatch(logout());  //limpiar auth y localStorage
    }
}


const checkingFinish = () => ({ // Action para terminar el checking del token
    type: types.authCheckingFinish
})

const login = (user) => ({ // Action para login
    type: types.authLogin,
    payload: user
})

const logout = () => ({
    type: types.authLogout
})
