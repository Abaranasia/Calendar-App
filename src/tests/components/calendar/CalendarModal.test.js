import React from 'react';
import { mount } from 'enzyme';
import { Provider } from 'react-redux';
import moment from 'moment';

import configureStore from 'redux-mock-store';
import thunk from 'redux-thunk';
import '@testing-library/jest-dom';
import { act } from 'react-dom/test-utils';

import Swal from 'sweetalert2';

import { CalendarModal } from '../../../components/calendar/CalendarModal';
import { eventStartUpdate, eventClearActiveNote, eventStartAddNew } from '../../../actions/events';


jest.mock('../../../actions/events', () => ({
    eventStartUpdate: jest.fn(),
    eventStartAddNew: jest.fn(),
    eventClearActiveNote: jest.fn(), //Para evitar que pete el anterior
}));


const middlewares = [thunk];
const mockStore = configureStore(middlewares);


const now = moment().minutes(0).seconds(0).add(1, 'hours'); // Esto establece el momento inicial no ahora si no a la próxima hora punta posible
const later = now.clone().add(1, 'hours'); //función de moment que clona otro momento


const initState = {
    auth: {
        uid: '123',
        name: 'testuser'
    },
    calendar: {
        events: [],
        activeEvent: {
            title: 'Evento',
            notes: 'prueba',
            start: now.toDate(),
            end: later.toDate()
        }
    },
    ui: {
        modalOpen: true
    }
};


const store = mockStore(initState);
store.dispatch = jest.fn();

const wrapper = mount(
    <Provider store={store}>
        <CalendarModal />
    </Provider>
)


describe('<CalendarModal/> tests', () => {

    test('should display the modal component properly', () => {

        expect(wrapper.find('.modal').exists()).toBeTruthy()
        expect(wrapper.find('Modal').prop('isOpen')).toBeTruthy()
    });

    test('should call the update action and close modal', () => {

        wrapper.find('form').simulate('submit', {
            preventDefault() { }
        })

        expect(eventStartUpdate).toHaveBeenCalledWith(initState.calendar.activeEvent);
        expect(eventClearActiveNote).toHaveBeenCalled();
    });



    test('should display an error if there is no title in the form', () => {
        // Esta prueba hace submit con el form vacío porque en la anterior se cerró y debió limpiarse (aunque no y lo forzamos)

        //wrapper.find('input[name="title"]').value = '';
        wrapper.find('input[name="title"]').simulate('change', {
            target: {
                name: 'title', value: ''
            }
        });

        wrapper.find('form').simulate('submit', {
            preventDefault() { }
        });

        expect(wrapper).toMatchSnapshot();
        expect(wrapper.find('input[name="title"]').hasClass('is-invalid')).toBeTruthy()
    });


    test('should create a new event', () => {

        const initState = {
            auth: {
                uid: '123',
                name: 'testuser'
            },
            calendar: {
                events: [],
                activeEvent: null
            },
            ui: {
                modalOpen: true
            }
        };

        const store = mockStore(initState);
        store.dispatch = jest.fn();

        const wrapper = mount(
            <Provider store={store}>
                <CalendarModal />
            </Provider>
        )

        wrapper.find('input[name="title"]').simulate('change', {
            target: {
                name: 'title', value: 'hola pruebas'
            }
        });

        wrapper.find('form').simulate('submit', {
            preventDefault() { }
        });

        expect(eventStartAddNew).toHaveBeenCalledWith({
            end: expect.anything(),
            start: expect.anything(),
            title: 'hola pruebas',
            notes: 'prueba'
        });

        expect(eventClearActiveNote).toHaveBeenCalled();
    });


    test('dates should be validated ', () => {

        wrapper.find('input[name="title"]').simulate('change', {
            target: {
                name: 'title', value: 'hola pruebas'
            }
        });

        const hoy = new Date();

        act(() => {
            wrapper.find('DateTimePicker').at(1).prop('onChange')(hoy) // .at(1) en referencia al segundo DateTimePicker, el de fin del evento
        });

        wrapper.find('form').simulate('submit', {
            preventDefault() { }
        });

        expect(Swal.fire).toHaveBeenCalledWith("Error", "End date should be later than start date", "error")
    });
})
