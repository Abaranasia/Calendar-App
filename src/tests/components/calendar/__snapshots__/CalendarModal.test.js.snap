// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<CalendarModal/> tests should display an error if there is no title in the form 1`] = `
<Portal
  containerInfo={
    <div
      class="ReactModalPortal"
    >
      <div
        class="ReactModal__Overlay modal-fondo"
      >
        <div
          aria-modal="true"
          class="ReactModal__Content modal"
          role="dialog"
          style="top: 50%; left: 50%; margin-right: -50%; transform: translate(-50%, -50%);"
          tabindex="-1"
        >
          <h2>
             Nuevo evento 
          </h2>
          <hr />
          <form
            class="container"
          >
            <div
              class="form-group"
            >
              <label>
                Fecha y hora inicio
              </label>
              <div
                class="react-datetime-picker react-datetime-picker--closed react-datetime-picker--enabled form-control"
              >
                <div
                  class="react-datetime-picker__wrapper"
                >
                  <div
                    class="react-datetime-picker__inputGroup"
                  >
                    <input
                      max="275760-09-13T02:00"
                      min="0001-01-01T00:00"
                      name="datetime"
                      step="60"
                      style="visibility: hidden; position: absolute; z-index: -999;"
                      type="datetime-local"
                      value="2021-11-30T00:00"
                    />
                    
                    <input
                      autocomplete="off"
                      class="react-datetime-picker__inputGroup__input react-datetime-picker__inputGroup__month"
                      data-input="true"
                      inputmode="numeric"
                      max="12"
                      min="1"
                      name="month"
                      placeholder="--"
                      type="number"
                      value="11"
                    />
                    <span
                      class="react-datetime-picker__inputGroup__divider"
                    >
                      /
                    </span>
                    <input
                      autocomplete="off"
                      class="react-datetime-picker__inputGroup__input react-datetime-picker__inputGroup__day"
                      data-input="true"
                      inputmode="numeric"
                      max="30"
                      min="1"
                      name="day"
                      placeholder="--"
                      type="number"
                      value="30"
                    />
                    <span
                      class="react-datetime-picker__inputGroup__divider"
                    >
                      /
                    </span>
                    <input
                      autocomplete="off"
                      class="react-datetime-picker__inputGroup__input react-datetime-picker__inputGroup__year"
                      data-input="true"
                      inputmode="numeric"
                      max="275760"
                      min="1"
                      name="year"
                      placeholder="----"
                      step="1"
                      type="number"
                      value="2021"
                    />
                    <span
                      class="react-datetime-picker__inputGroup__divider"
                    >
                      Â 
                    </span>
                    <input
                      autocomplete="off"
                      class="react-datetime-picker__inputGroup__input react-datetime-picker__inputGroup__hour"
                      data-input="true"
                      inputmode="numeric"
                      max="12"
                      min="1"
                      name="hour12"
                      placeholder="--"
                      type="number"
                      value="12"
                    />
                    <span
                      class="react-datetime-picker__inputGroup__divider"
                    >
                      :
                    </span>
                    <span
                      class="react-datetime-picker__inputGroup__leadingZero"
                    >
                      0
                    </span>
                    <input
                      autocomplete="off"
                      class="react-datetime-picker__inputGroup__input react-datetime-picker__inputGroup__minute react-datetime-picker__inputGroup__input--hasLeadingZero"
                      data-input="true"
                      inputmode="numeric"
                      max="59"
                      min="0"
                      name="minute"
                      placeholder="--"
                      type="number"
                      value="0"
                    />
                    <span
                      class="react-datetime-picker__inputGroup__divider"
                    >
                       
                    </span>
                    <select
                      class="react-datetime-picker__inputGroup__input react-datetime-picker__inputGroup__amPm"
                      data-input="true"
                      data-select="true"
                      name="amPm"
                    >
                      <option
                        value="am"
                      >
                        AM
                      </option>
                      <option
                        value="pm"
                      >
                        PM
                      </option>
                    </select>
                    
                  </div>
                  <button
                    class="react-datetime-picker__clear-button react-datetime-picker__button"
                    type="button"
                  >
                    <svg
                      class="react-datetime-picker__clear-button__icon react-datetime-picker__button__icon"
                      height="19"
                      stroke="black"
                      stroke-width="2"
                      viewBox="0 0 19 19"
                      width="19"
                      xmlns="http://www.w3.org/2000/svg"
                    >
                      <line
                        x1="4"
                        x2="15"
                        y1="4"
                        y2="15"
                      />
                      <line
                        x1="15"
                        x2="4"
                        y1="4"
                        y2="15"
                      />
                    </svg>
                  </button>
                  <button
                    class="react-datetime-picker__calendar-button react-datetime-picker__button"
                    type="button"
                  >
                    <svg
                      class="react-datetime-picker__calendar-button__icon react-datetime-picker__button__icon"
                      height="19"
                      stroke="black"
                      stroke-width="2"
                      viewBox="0 0 19 19"
                      width="19"
                      xmlns="http://www.w3.org/2000/svg"
                    >
                      <rect
                        fill="none"
                        height="15"
                        width="15"
                        x="2"
                        y="2"
                      />
                      <line
                        x1="6"
                        x2="6"
                        y1="0"
                        y2="4"
                      />
                      <line
                        x1="13"
                        x2="13"
                        y1="0"
                        y2="4"
                      />
                    </svg>
                  </button>
                </div>
              </div>
            </div>
            <div
              class="form-group"
            >
              <label>
                Fecha y hora fin
              </label>
              <div
                class="react-datetime-picker react-datetime-picker--closed react-datetime-picker--enabled form-control"
              >
                <div
                  class="react-datetime-picker__wrapper"
                >
                  <div
                    class="react-datetime-picker__inputGroup"
                  >
                    <input
                      max="275760-09-13T02:00"
                      min="2021-11-30T00:00"
                      name="datetime"
                      step="60"
                      style="visibility: hidden; position: absolute; z-index: -999;"
                      type="datetime-local"
                      value="2021-11-30T01:00"
                    />
                    
                    <input
                      autocomplete="off"
                      class="react-datetime-picker__inputGroup__input react-datetime-picker__inputGroup__month"
                      data-input="true"
                      inputmode="numeric"
                      max="12"
                      min="11"
                      name="month"
                      placeholder="--"
                      type="number"
                      value="11"
                    />
                    <span
                      class="react-datetime-picker__inputGroup__divider"
                    >
                      /
                    </span>
                    <input
                      autocomplete="off"
                      class="react-datetime-picker__inputGroup__input react-datetime-picker__inputGroup__day"
                      data-input="true"
                      inputmode="numeric"
                      max="30"
                      min="30"
                      name="day"
                      placeholder="--"
                      type="number"
                      value="30"
                    />
                    <span
                      class="react-datetime-picker__inputGroup__divider"
                    >
                      /
                    </span>
                    <input
                      autocomplete="off"
                      class="react-datetime-picker__inputGroup__input react-datetime-picker__inputGroup__year"
                      data-input="true"
                      inputmode="numeric"
                      max="275760"
                      min="2021"
                      name="year"
                      placeholder="----"
                      step="1"
                      type="number"
                      value="2021"
                    />
                    <span
                      class="react-datetime-picker__inputGroup__divider"
                    >
                      Â 
                    </span>
                    <input
                      autocomplete="off"
                      class="react-datetime-picker__inputGroup__input react-datetime-picker__inputGroup__hour"
                      data-input="true"
                      inputmode="numeric"
                      max="12"
                      min="1"
                      name="hour12"
                      placeholder="--"
                      type="number"
                      value="1"
                    />
                    <span
                      class="react-datetime-picker__inputGroup__divider"
                    >
                      :
                    </span>
                    <span
                      class="react-datetime-picker__inputGroup__leadingZero"
                    >
                      0
                    </span>
                    <input
                      autocomplete="off"
                      class="react-datetime-picker__inputGroup__input react-datetime-picker__inputGroup__minute react-datetime-picker__inputGroup__input--hasLeadingZero"
                      data-input="true"
                      inputmode="numeric"
                      max="59"
                      min="0"
                      name="minute"
                      placeholder="--"
                      type="number"
                      value="0"
                    />
                    <span
                      class="react-datetime-picker__inputGroup__divider"
                    >
                       
                    </span>
                    <select
                      class="react-datetime-picker__inputGroup__input react-datetime-picker__inputGroup__amPm"
                      data-input="true"
                      data-select="true"
                      name="amPm"
                    >
                      <option
                        value="am"
                      >
                        AM
                      </option>
                      <option
                        value="pm"
                      >
                        PM
                      </option>
                    </select>
                    
                  </div>
                  <button
                    class="react-datetime-picker__clear-button react-datetime-picker__button"
                    type="button"
                  >
                    <svg
                      class="react-datetime-picker__clear-button__icon react-datetime-picker__button__icon"
                      height="19"
                      stroke="black"
                      stroke-width="2"
                      viewBox="0 0 19 19"
                      width="19"
                      xmlns="http://www.w3.org/2000/svg"
                    >
                      <line
                        x1="4"
                        x2="15"
                        y1="4"
                        y2="15"
                      />
                      <line
                        x1="15"
                        x2="4"
                        y1="4"
                        y2="15"
                      />
                    </svg>
                  </button>
                  <button
                    class="react-datetime-picker__calendar-button react-datetime-picker__button"
                    type="button"
                  >
                    <svg
                      class="react-datetime-picker__calendar-button__icon react-datetime-picker__button__icon"
                      height="19"
                      stroke="black"
                      stroke-width="2"
                      viewBox="0 0 19 19"
                      width="19"
                      xmlns="http://www.w3.org/2000/svg"
                    >
                      <rect
                        fill="none"
                        height="15"
                        width="15"
                        x="2"
                        y="2"
                      />
                      <line
                        x1="6"
                        x2="6"
                        y1="0"
                        y2="4"
                      />
                      <line
                        x1="13"
                        x2="13"
                        y1="0"
                        y2="4"
                      />
                    </svg>
                  </button>
                </div>
              </div>
            </div>
            <hr />
            <div
              class="form-group"
            >
              <label>
                Titulo y notas
              </label>
              <input
                autocomplete="off"
                class="form-control is-invalid"
                name="title"
                placeholder="TÃ­tulo del evento"
                type="text"
                value=""
              />
              <small
                class="form-text text-muted"
                id="emailHelp"
              >
                Una descripciÃ³n corta
              </small>
            </div>
            <div
              class="form-group"
            >
              <textarea
                class="form-control"
                name="notes"
                placeholder="Notas"
                rows="5"
                type="text"
              >
                prueba
              </textarea>
              <small
                class="form-text text-muted"
                id="emailHelp"
              >
                InformaciÃ³n adicional
              </small>
            </div>
            <button
              class="btn btn-outline-primary btn-block"
              type="submit"
            >
              <i
                class="far fa-save"
              />
              <span>
                 Guardar
              </span>
            </button>
          </form>
        </div>
      </div>
    </div>
  }
>
  <div
    className="ReactModal__Overlay modal-fondo"
    onClick={[Function]}
    onMouseDown={[Function]}
    style={Object {}}
  >
    <div
      aria-modal={true}
      className="ReactModal__Content modal"
      onClick={[Function]}
      onKeyDown={[Function]}
      onMouseDown={[Function]}
      onMouseUp={[Function]}
      role="dialog"
      style={
        Object {
          "bottom": "auto",
          "left": "50%",
          "marginRight": "-50%",
          "right": "auto",
          "top": "50%",
          "transform": "translate(-50%, -50%)",
        }
      }
      tabIndex="-1"
    >
      <h2>
         Nuevo evento 
      </h2>
      <hr />
      <form
        className="container"
        onSubmit={[Function]}
      >
        <div
          className="form-group"
        >
          <label>
            Fecha y hora inicio
          </label>
          <div
            className="react-datetime-picker react-datetime-picker--closed react-datetime-picker--enabled form-control"
            onFocus={[Function]}
          >
            <div
              className="react-datetime-picker__wrapper"
            >
              <div
                className="react-datetime-picker__inputGroup"
                onClick={[Function]}
              >
                <input
                  max="275760-09-13T02:00"
                  min="0001-01-01T00:00"
                  name="datetime"
                  onChange={[Function]}
                  onFocus={[Function]}
                  step={60}
                  style={
                    Object {
                      "position": "absolute",
                      "visibility": "hidden",
                      "zIndex": "-999",
                    }
                  }
                  type="datetime-local"
                  value="2021-11-30T00:00"
                />
                <input
                  autoComplete="off"
                  className="react-datetime-picker__inputGroup__input react-datetime-picker__inputGroup__month"
                  data-input="true"
                  inputMode="numeric"
                  key="input"
                  max={12}
                  min={1}
                  name="month"
                  onChange={[Function]}
                  onFocus={[Function]}
                  onKeyDown={[Function]}
                  onKeyPress={[Function]}
                  onKeyUp={[Function]}
                  placeholder="--"
                  required={null}
                  type="number"
                  value="11"
                />
                <span
                  className="react-datetime-picker__inputGroup__divider"
                >
                  /
                </span>
                <input
                  autoComplete="off"
                  autoFocus={false}
                  className="react-datetime-picker__inputGroup__input react-datetime-picker__inputGroup__day"
                  data-input="true"
                  inputMode="numeric"
                  key="input"
                  max={30}
                  min={1}
                  name="day"
                  onChange={[Function]}
                  onFocus={[Function]}
                  onKeyDown={[Function]}
                  onKeyPress={[Function]}
                  onKeyUp={[Function]}
                  placeholder="--"
                  required={null}
                  type="number"
                  value="30"
                />
                <span
                  className="react-datetime-picker__inputGroup__divider"
                >
                  /
                </span>
                <input
                  autoComplete="off"
                  autoFocus={false}
                  className="react-datetime-picker__inputGroup__input react-datetime-picker__inputGroup__year"
                  data-input="true"
                  inputMode="numeric"
                  key="input"
                  max={275760}
                  min={1}
                  name="year"
                  onChange={[Function]}
                  onFocus={[Function]}
                  onKeyDown={[Function]}
                  onKeyPress={[Function]}
                  onKeyUp={[Function]}
                  placeholder="----"
                  required={null}
                  step={1}
                  type="number"
                  value="2021"
                />
                <span
                  className="react-datetime-picker__inputGroup__divider"
                >
                  Â 
                </span>
                <input
                  autoComplete="off"
                  autoFocus={false}
                  className="react-datetime-picker__inputGroup__input react-datetime-picker__inputGroup__hour"
                  data-input="true"
                  inputMode="numeric"
                  key="input"
                  max={12}
                  min={1}
                  name="hour12"
                  onChange={[Function]}
                  onFocus={[Function]}
                  onKeyDown={[Function]}
                  onKeyPress={[Function]}
                  onKeyUp={[Function]}
                  placeholder="--"
                  required={null}
                  type="number"
                  value="12"
                />
                <span
                  className="react-datetime-picker__inputGroup__divider"
                >
                  :
                </span>
                Array [
                  <span
                    className="react-datetime-picker__inputGroup__leadingZero"
                    key="leadingZero"
                  >
                    0
                  </span>,
                  <input
                    autoComplete="off"
                    autoFocus={false}
                    className="react-datetime-picker__inputGroup__input react-datetime-picker__inputGroup__minute react-datetime-picker__inputGroup__input--hasLeadingZero"
                    data-input="true"
                    inputMode="numeric"
                    key="input"
                    max={59}
                    min={0}
                    name="minute"
                    onChange={[Function]}
                    onFocus={[Function]}
                    onKeyDown={[Function]}
                    onKeyPress={[Function]}
                    onKeyUp={[Function]}
                    placeholder="--"
                    required={null}
                    type="number"
                    value="0"
                  />,
                ]
                <span
                  className="react-datetime-picker__inputGroup__divider"
                >
                   
                </span>
                <select
                  className="react-datetime-picker__inputGroup__input react-datetime-picker__inputGroup__amPm"
                  data-input="true"
                  data-select="true"
                  name="amPm"
                  onChange={[Function]}
                  required={null}
                  value="am"
                >
                  <option
                    value="am"
                  >
                    AM
                  </option>
                  <option
                    value="pm"
                  >
                    PM
                  </option>
                </select>
              </div>
              <button
                className="react-datetime-picker__clear-button react-datetime-picker__button"
                onClick={[Function]}
                onFocus={[Function]}
                type="button"
              >
                <svg
                  className="react-datetime-picker__clear-button__icon react-datetime-picker__button__icon"
                  height={19}
                  stroke="black"
                  strokeWidth={2}
                  viewBox="0 0 19 19"
                  width={19}
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <line
                    x1="4"
                    x2="15"
                    y1="4"
                    y2="15"
                  />
                  <line
                    x1="15"
                    x2="4"
                    y1="4"
                    y2="15"
                  />
                </svg>
              </button>
              <button
                className="react-datetime-picker__calendar-button react-datetime-picker__button"
                onClick={[Function]}
                onFocus={[Function]}
                type="button"
              >
                <svg
                  className="react-datetime-picker__calendar-button__icon react-datetime-picker__button__icon"
                  height={19}
                  stroke="black"
                  strokeWidth={2}
                  viewBox="0 0 19 19"
                  width={19}
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <rect
                    fill="none"
                    height="15"
                    width="15"
                    x="2"
                    y="2"
                  />
                  <line
                    x1="6"
                    x2="6"
                    y1="0"
                    y2="4"
                  />
                  <line
                    x1="13"
                    x2="13"
                    y1="0"
                    y2="4"
                  />
                </svg>
              </button>
            </div>
          </div>
        </div>
        <div
          className="form-group"
        >
          <label>
            Fecha y hora fin
          </label>
          <div
            className="react-datetime-picker react-datetime-picker--closed react-datetime-picker--enabled form-control"
            onFocus={[Function]}
          >
            <div
              className="react-datetime-picker__wrapper"
            >
              <div
                className="react-datetime-picker__inputGroup"
                onClick={[Function]}
              >
                <input
                  max="275760-09-13T02:00"
                  min="2021-11-30T00:00"
                  name="datetime"
                  onChange={[Function]}
                  onFocus={[Function]}
                  step={60}
                  style={
                    Object {
                      "position": "absolute",
                      "visibility": "hidden",
                      "zIndex": "-999",
                    }
                  }
                  type="datetime-local"
                  value="2021-11-30T01:00"
                />
                <input
                  autoComplete="off"
                  className="react-datetime-picker__inputGroup__input react-datetime-picker__inputGroup__month"
                  data-input="true"
                  inputMode="numeric"
                  key="input"
                  max={12}
                  min={11}
                  name="month"
                  onChange={[Function]}
                  onFocus={[Function]}
                  onKeyDown={[Function]}
                  onKeyPress={[Function]}
                  onKeyUp={[Function]}
                  placeholder="--"
                  required={null}
                  type="number"
                  value="11"
                />
                <span
                  className="react-datetime-picker__inputGroup__divider"
                >
                  /
                </span>
                <input
                  autoComplete="off"
                  autoFocus={false}
                  className="react-datetime-picker__inputGroup__input react-datetime-picker__inputGroup__day"
                  data-input="true"
                  inputMode="numeric"
                  key="input"
                  max={30}
                  min={30}
                  name="day"
                  onChange={[Function]}
                  onFocus={[Function]}
                  onKeyDown={[Function]}
                  onKeyPress={[Function]}
                  onKeyUp={[Function]}
                  placeholder="--"
                  required={null}
                  type="number"
                  value="30"
                />
                <span
                  className="react-datetime-picker__inputGroup__divider"
                >
                  /
                </span>
                <input
                  autoComplete="off"
                  autoFocus={false}
                  className="react-datetime-picker__inputGroup__input react-datetime-picker__inputGroup__year"
                  data-input="true"
                  inputMode="numeric"
                  key="input"
                  max={275760}
                  min={2021}
                  name="year"
                  onChange={[Function]}
                  onFocus={[Function]}
                  onKeyDown={[Function]}
                  onKeyPress={[Function]}
                  onKeyUp={[Function]}
                  placeholder="----"
                  required={null}
                  step={1}
                  type="number"
                  value="2021"
                />
                <span
                  className="react-datetime-picker__inputGroup__divider"
                >
                  Â 
                </span>
                <input
                  autoComplete="off"
                  autoFocus={false}
                  className="react-datetime-picker__inputGroup__input react-datetime-picker__inputGroup__hour"
                  data-input="true"
                  inputMode="numeric"
                  key="input"
                  max={12}
                  min={1}
                  name="hour12"
                  onChange={[Function]}
                  onFocus={[Function]}
                  onKeyDown={[Function]}
                  onKeyPress={[Function]}
                  onKeyUp={[Function]}
                  placeholder="--"
                  required={null}
                  type="number"
                  value="1"
                />
                <span
                  className="react-datetime-picker__inputGroup__divider"
                >
                  :
                </span>
                Array [
                  <span
                    className="react-datetime-picker__inputGroup__leadingZero"
                    key="leadingZero"
                  >
                    0
                  </span>,
                  <input
                    autoComplete="off"
                    autoFocus={false}
                    className="react-datetime-picker__inputGroup__input react-datetime-picker__inputGroup__minute react-datetime-picker__inputGroup__input--hasLeadingZero"
                    data-input="true"
                    inputMode="numeric"
                    key="input"
                    max={59}
                    min={0}
                    name="minute"
                    onChange={[Function]}
                    onFocus={[Function]}
                    onKeyDown={[Function]}
                    onKeyPress={[Function]}
                    onKeyUp={[Function]}
                    placeholder="--"
                    required={null}
                    type="number"
                    value="0"
                  />,
                ]
                <span
                  className="react-datetime-picker__inputGroup__divider"
                >
                   
                </span>
                <select
                  className="react-datetime-picker__inputGroup__input react-datetime-picker__inputGroup__amPm"
                  data-input="true"
                  data-select="true"
                  name="amPm"
                  onChange={[Function]}
                  required={null}
                  value="am"
                >
                  <option
                    value="am"
                  >
                    AM
                  </option>
                  <option
                    value="pm"
                  >
                    PM
                  </option>
                </select>
              </div>
              <button
                className="react-datetime-picker__clear-button react-datetime-picker__button"
                onClick={[Function]}
                onFocus={[Function]}
                type="button"
              >
                <svg
                  className="react-datetime-picker__clear-button__icon react-datetime-picker__button__icon"
                  height={19}
                  stroke="black"
                  strokeWidth={2}
                  viewBox="0 0 19 19"
                  width={19}
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <line
                    x1="4"
                    x2="15"
                    y1="4"
                    y2="15"
                  />
                  <line
                    x1="15"
                    x2="4"
                    y1="4"
                    y2="15"
                  />
                </svg>
              </button>
              <button
                className="react-datetime-picker__calendar-button react-datetime-picker__button"
                onClick={[Function]}
                onFocus={[Function]}
                type="button"
              >
                <svg
                  className="react-datetime-picker__calendar-button__icon react-datetime-picker__button__icon"
                  height={19}
                  stroke="black"
                  strokeWidth={2}
                  viewBox="0 0 19 19"
                  width={19}
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <rect
                    fill="none"
                    height="15"
                    width="15"
                    x="2"
                    y="2"
                  />
                  <line
                    x1="6"
                    x2="6"
                    y1="0"
                    y2="4"
                  />
                  <line
                    x1="13"
                    x2="13"
                    y1="0"
                    y2="4"
                  />
                </svg>
              </button>
            </div>
          </div>
        </div>
        <hr />
        <div
          className="form-group"
        >
          <label>
            Titulo y notas
          </label>
          <input
            autoComplete="off"
            className="form-control is-invalid"
            name="title"
            onChange={[Function]}
            placeholder="TÃ­tulo del evento"
            type="text"
            value=""
          />
          <small
            className="form-text text-muted"
            id="emailHelp"
          >
            Una descripciÃ³n corta
          </small>
        </div>
        <div
          className="form-group"
        >
          <textarea
            className="form-control"
            name="notes"
            onChange={[Function]}
            placeholder="Notas"
            rows="5"
            type="text"
            value="prueba"
          />
          <small
            className="form-text text-muted"
            id="emailHelp"
          >
            InformaciÃ³n adicional
          </small>
        </div>
        <button
          className="btn btn-outline-primary btn-block"
          type="submit"
        >
          <i
            className="far fa-save"
          />
          <span>
             Guardar
          </span>
        </button>
      </form>
    </div>
  </div>
</Portal>
`;
